src/metadata.js

```diff
@@ -75,17 +75,24 @@ Metadata.getGAEDescriptor = function() {
  * @return {object}
  */
 Metadata.getGCEDescriptor = function(callback) {
-  gcpMetadata
-    .instance('id')
-    .then(function(resp) {
+  gcpMetadata.instance('id').then(function(idResponse) {
+    gcpMetadata.instance('zone').then(function(zoneResponse) {
+      // Some parsing is necessary. Metadata service returns a fully
+      // qualified zone name: 'projects/{projectId}/zones/{zone}'. Logging
+      // wants just the zone part.
+      //
+      // eslint-disable-next-line no-unused-vars
+      const [_, zone] =
```

Comment:
This might be more simple, and follows what we usually do in ServiceObject constructors to determine if the user passed us a full resource path, or just the last part:

```js
const zone = zoneResponse.data.split('/').pop();
```

Reply:
👍